pyteal/ast/abi/util_test.py

```diff
@@ -772,3 +772,250 @@ def test_sdk_type_specs_from_signature(sig_str, sig_args, sig_rets):
     args, ret = abi.type_specs_from_signature(sig_str)
     assert args == sig_args
     assert ret == sig_rets
+
+
+class NamedTDecl(abi.NamedTuple):
+    a: abi.Field[abi.Uint64]
+    b: abi.Field[
+        abi.Tuple3[abi.PaymentTransaction, abi.Address, abi.StaticBytes[Literal[16]]]
+    ]
+    c: abi.Field[abi.Transaction]
+
+
+class NamedTComp0(abi.NamedTuple):
+    a0: abi.Field[abi.String]
+    a1: abi.Field[abi.Address]
+
+
+class NamedTComp1(abi.NamedTuple):
+    b0: abi.Field[abi.DynamicBytes]
+    b1: abi.Field[abi.StaticBytes[Literal[32]]]
+
+
+class NamedTComp2(abi.NamedTuple):
+    b1: abi.Field[abi.Address]
+    b0: abi.Field[abi.DynamicBytes]
+
+
+class SafeBidirectional(NamedTuple):
+    xs: list[abi.TypeSpec]
+
+
+SAFE_BIDIRECTIONAL_TEST_CASES: list[SafeBidirectional] = (
```

Comment:
@ahangsu Testing my understanding:  Should the following test case pass?  

It fails when I tested locally.  If you think it should fail, can you help me understand the intuition?  I think the encoding is equivalent and it feels like an alias to me.

`SafeBidirectional([abi.ByteTypeSpec(), abi.Uint8TypeSpec()]),`

Reply:
This feels like something worth discussing. When we started working on ABI, even before `TypeSpec` was introduced, we did want to make `byte` inherit from `byte`, see #194.

Later that approach did not work out, and in PR #222 we introduced `TypeSpec`, but the inheritance of `byte` from `uint8` was removed, in this commit: https://github.com/algorand/pyteal/pull/222/commits/4fe7dbf1bf24c1e1b30a96291e9f0e308f70f8d0.

My thinking is, we probably should not change the implementation on `uint` side, but add a rule over `uint`, such that 2 typespecs are mutually assignable if they have same byte size.

I would also like to consult @jasonpaulos for his opinion, in case he sees some drawbacks that we failed to see.

Reply:
I pushed the change to reflect that the `uint8` and `byte` are mutual assignable, let me know if we should roll back.

Reply:
@ahangsu Thanks for the change in https://github.com/algorand/pyteal/pull/540/commits/3256254417b66dcc8c2a3a353e196207cc8fea71.  It matches my expectation.  

@jasonpaulos Leaving the thread open for you to sanity check - Can you confirm you're on-board?